x/supplier/types/message_submit_proof.go

```diff
@@ -40,10 +41,42 @@ func (msg *MsgSubmitProof) GetSignBytes() []byte {
 	return sdk.MustSortJSON(bz)
 }
 
+// ValidateBasic ensures that the bech32 address strings for the supplier and
+// application addresses are valid and that the proof and service ID are not empty.
+//
+// TODO_CONSIDERATION: additional assertions:
+// * session ID is not empty
+// * session end - start height == on-chain NumBlocksPerSession (for that session)
 func (msg *MsgSubmitProof) ValidateBasic() error {
-	_, err := sdk.AccAddressFromBech32(msg.SupplierAddress)
+	var errMsg string
+	_, err := sdk.AccAddressFromBech32(msg.GetSupplierAddress())
 	if err != nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid supplierAddress address (%s)", err)
+		return sdkerrors.ErrInvalidAddress.Wrapf(
+			"supplier address %q, error: %s",
+			msg.GetSupplierAddress(),
+			err,
+		)
 	}
+
+	_, err = sdk.AccAddressFromBech32(msg.GetSessionHeader().GetApplicationAddress())
+	if err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrapf(
+			"application address: %q, error: %s",
+			msg.GetSessionHeader().GetApplicationAddress(),
+			err,
+		)
+	}
+
+	if msg.GetSessionHeader().GetService().GetId() == "" {
+		return ErrSupplierInvalidService.Wrap("proof service ID %q cannot be empty")
+	}
+
+	if len(msg.GetProof()) == 0 {
```

Comment:
Should we try to parse it (using the proto) as well? Similar to how we do address validation?

Feel free to add a TODO.